scripts/entrypoint.sh

```diff
@@ -35,13 +35,54 @@ log_in_json()
     printf '{"level":"%s","ts":"%s","caller":"%s","msg":"%s"}\n' "$LOGTYPE" "$TIMESTAMP" "$FILENAME" "$MSG"
 }
 
+unsupported_prefix_target_conf()
+{
+   if [ "${WARM_PREFIX_TARGET}" == "0" ];then
+        true
+   else
+        false
+    fi    
+}
+
+warm_ip_target_not_configured()
+{
+    if [ "${WARM_IP_TARGET}" == "0" ];then
+        true
+    else
+        false
+    fi    
+}
+
+min_ip_target_not_configured()
+{
+    if [ "${MINIMUM_IP_TARGET}" == "0" ]; then
+        true
+    else
+        false
+    fi    
+}
+
+is_prefix_delegation_enabled()
+{
+    if [ "${ENABLE_PREFIX_DELEGATION}" == "true" ]; then
+        true
+    else
+        false
+    fi    
+}
+
 validate_env_var()
 {
     log_in_json info "Validating env variables ..."
     if [[ "${AWS_VPC_K8S_PLUGIN_LOG_FILE,,}" == "stdout" ]]; then
         log_in_json error "AWS_VPC_K8S_PLUGIN_LOG_FILE cannot be set to stdout"
         exit 1     
     fi 
+
+    if is_prefix_delegation_enabled && unsupported_prefix_target_conf && warm_ip_target_not_configured && min_ip_target_not_configured ; then
```

Comment:
can we merge this into a single function instead?
like validate_warm_prefix_conf

we added 4 un-necessary functions to support purpose, while these checks can be simply lined in a single function, and reads better.

Reply:
Yes initially I had it one line but mainly for readability I did this.